typing/types.mli

```diff
@@ -625,7 +625,7 @@ type module_type =
 
 and functor_parameter =
   | Unit
-  | Named of Ident.t option * module_type
+  | Named of bool * Ident.t option * module_type
```

Comment:
This probably shouldn't be a boolean.
I think that this should either be a 'purity' type or this case should be turned into an inline record